Add support for ODBC pollers in Zabbix >= 6.0

fraenki opened the pull request:
<!--
Thank you for contributing to this project!

- This project has a Contributor Code of Conduct: https://voxpupuli.org/coc/
- Please check that here is no existing issue or PR that addresses your problem.
- Our vulnerabilities reporting process is at https://voxpupuli.org/security/

-->
#### Pull Request (PR) description

This PR adds support for the `StartODBCPollers` configuration parameter, that was introduced in Zabbix 6.0.

#### This Pull Request (PR) fixes the following issues

This parameter was already discussed in https://github.com/voxpupuli/puppet-zabbix/issues/814#issuecomment-1042347988, but a PR was never submitted, AFAICT.

fraenki commented:
The test failure does not seem to be related to this PR.

smortex commented:
> REFERENCE.md is outdated

Please update REFERENCE.md with

```sh-session
$ bundle exec rake strings:generate:reference
```

Add it to the PR and CI should continue.

fraenki commented:
> Please update REFERENCE.md with

Done.

root-expert commented:
@fraenki Please consider adding basic tests for the new parameter both for the proxy and the server :smile:

fraenki commented:
@root-expert Rebased against current master and added basic tests.

(Side note: This module runs most tests with zabbix_version=5.0 and zabbix_version=6.0 is not supported for the server class on all operating systems. As a result the tests I've added aren't perfect. But once all tests run with zabbix_version=6.0 everything should be fine.)

root-expert commented on manifests/proxy.pp:
```diff
@@ -237,6 +238,7 @@
   $datasenderfrequency                                                        = $zabbix::params::proxy_datasenderfrequency,
   $startpollers                                                               = $zabbix::params::proxy_startpollers,
   $startipmipollers                                                           = $zabbix::params::proxy_startipmipollers,
+  $startodbcpollers                                                           = $zabbix::params::proxy_startodbcpollers,
```
```suggestion
  Integer[0, 1000] $startodbcpollers                                          = $zabbix::params::proxy_startodbcpollers,
```

root-expert commented on manifests/server.pp:
```diff
@@ -202,6 +203,7 @@
   Optional[String[1]] $database_tlscipher13                                   = $zabbix::params::server_database_tlscipher13,
   $startpollers                                                               = $zabbix::params::server_startpollers,
   $startipmipollers                                                           = $zabbix::params::server_startipmipollers,
+  $startodbcpollers                                                           = $zabbix::params::server_startodbcpollers,
```
```suggestion
  Integer[0, 1000] $startodbcpollers                                          = $zabbix::params::server_startodbcpollers,
```

root-expert commented on spec/classes/proxy_spec.rb:
```diff
@@ -366,11 +366,26 @@
           let :params do
             {
               socketdir: '/var/run/zabbix',
+              startodbcpollers: '1',
```
```suggestion
              startodbcpollers: 1,
```

root-expert commented on spec/classes/proxy_spec.rb:
```diff
@@ -366,11 +366,26 @@
+          let :params do
+            {
+              socketdir: '/var/run/zabbix',
+              startodbcpollers: '1',
```
```suggestion
              startodbcpollers: 1,
```

root-expert commented on spec/classes/server_spec.rb:
```diff
@@ -367,11 +367,13 @@
         let :params do
           {
             socketdir: '/var/run/zabbix',
+            startodbcpollers: '1',
```
```suggestion
            startodbcpollers: 1,
```

root-expert commented on manifests/params.pp:
```diff
@@ -253,6 +253,7 @@
   $server_startipmipollers                  = '0'
   $server_startjavapollers                  = '5'
   $server_startlldprocessors                = 2
+  $server_startodbcpollers                  = '1'
```
```suggestion
  $server_startodbcpollers                  = 1
```

root-expert commented on manifests/params.pp:
```diff
@@ -422,6 +423,7 @@
   $proxy_startipmipollers                   = '0'
   $proxy_startjavapollers                   = '5'
   $proxy_startpingers                       = '1'
+  $proxy_startodbcpollers                   = '1'
```
```suggestion
  $proxy_startodbcpollers                   = 1
```

root-expert commented on manifests/init.pp:
```diff
@@ -274,6 +275,7 @@
   Optional[Stdlib::Absolutepath] $database_tlscafile                          = $zabbix::params::server_database_tlscafile,
   $startpollers                                                               = $zabbix::params::server_startpollers,
   $startipmipollers                                                           = $zabbix::params::server_startipmipollers,
+  $startodbcpollers                                                           = $zabbix::params::server_startodbcpollers,
```
```suggestion
  Integer[0, 1000] $startodbcpollers                                          = $zabbix::params::server_startodbcpollers,
```

fraenki commented:
@root-expert, all suggested changes applied, thanks. :)

evgeni commented on templates/zabbix_proxy.conf.erb:
```diff
@@ -356,6 +356,17 @@ UnavailableDelay=<%= @unavaliabledelay %>
 #
 UnreachableDelay=<%= @unreachabedelay %>
 
+## Option: StartODBCPollers
```
I'd place the `if @zabbix_version.to_f >= 6.0` before this line, as the comment doesn't make sense on 5.x either.

fraenki replied:
Done.

evgeni commented on templates/zabbix_server.conf.erb:
```diff
@@ -133,6 +133,17 @@ StartIPMIPollers=<%= @startipmipollers %>
 #
 StartPollersUnreachable=<%= @startpollersunreachable %>
 
+## Option: StartODBCPollers
```
I'd place the `if @zabbix_version.to_f >= 6.0` before this line, as the comment doesn't make sense on 5.x either.

fraenki replied:
Done.
